crates/types/src/traits/block_contents.rs

```diff
@@ -57,12 +57,18 @@ pub trait BlockPayload:
     /// If the transaction length conversion fails.
     fn from_transactions(
         transactions: impl IntoIterator<Item = Self::Transaction>,
+        num_storage_nodes: usize,
```

Comment:
Hmm. I don't think `BlockPayload` constructors need `num_storage_nodes`. We should talk about this on the next call, possibly with input from @jbearer . `BlockPayload` needs VID parameters (including `num_storage_nodes` plus some other stuff) but *only* for tx/namespace proofs.

Reply:
I agree, I think we can remove this if we get creative. The reason we have it here is because the constructor uses it to compute the VID commitment all of the transactions. Is this technically a _tx_ proof or am I thinking about it wrong?

Reply:
> Is this technically a _tx_ proof or am I thinking about it wrong?

I use "tx proof" to refer to something completely different. These proofs are used at the sequencer layer to service queries from light clients about individual txs or namespaces. It's not relevant to hotshot.

> The reason we have it here is because the constructor uses it to compute the VID commitment all of the transactions. 

The constructor does *not* compute the VID commitment of all the txs. The constructor merely marshals the txs into flat bytes, which can then be used by hotshot to compute the VID commitment.

Basically, `BlockPayload` trait allows the downstream user to specify how exactly txs are marshalled into bytes. But computing the actual VID commitment should be done only once, and it must be inside hotshot.

Reply:
Yeah, we originally had a `commit()` function on the `BlockPayload` trait, mostly for convenience. But we deleted it, because as Gus said it's not the responsibility of the payload itself to deal with VID. But I guess we just forgot to delete the `payload_commitment` field from the test payload type